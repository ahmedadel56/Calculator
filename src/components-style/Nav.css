.headerContainer {
  display: flex;
  justify-content: space-between;
  margin: 2rem 4rem;
}

.link {
  padding-right: 1rem;
  margin-left: 1rem;
  border-right: 3px solid black;
  color: blue;
}

.is-active {
  font-weight: bold;
}
