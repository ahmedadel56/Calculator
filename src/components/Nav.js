import React from 'react';
import { NavLink, BrowserRouter } from 'react-router-dom';
import '../components-style/Nav.css';

const Nav = () => (
  <header className="headerContainer" data-testid="Nav">
    <h1>Math Magicians</h1>
    <div>
      <BrowserRouter>
        <NavLink to="/" className="link" activeClassName="is-active" exact>
          Home
        </NavLink>
        <NavLink to="/calculator" className="link" activeClassName="is-active">
          Calculator
        </NavLink>
        <NavLink to="/quotes" className="link" activeClassName="is-active">
          Quotes
        </NavLink>
      </BrowserRouter>
    </div>
  </header>
);

export default Nav;
