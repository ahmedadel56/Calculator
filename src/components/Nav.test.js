import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Nav from './Nav';

describe('<Nav />', () => {
  test('it should mount', () => {
    render(<Nav />);

    const nav = screen.getByTestId('Nav');

    expect(nav).toBeInTheDocument();
  });

  it('it should match snapshot', () => {
    const tree = renderer.create(<Nav />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
