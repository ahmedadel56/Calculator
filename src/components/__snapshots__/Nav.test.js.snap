// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> it should match snapshot 1`] = `
<header
  className="headerContainer"
  data-testid="Nav"
>
  <h1>
    Math Magicians
  </h1>
  <div>
    <a
      aria-current="page"
      className="link is-active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Home
    </a>
    <a
      aria-current={null}
      className="link"
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      aria-current={null}
      className="link"
      href="/quotes"
      onClick={[Function]}
    >
      Quotes
    </a>
  </div>
</header>
`;
